### PR TITLE
Make the disc-capacity budget the build pipeline's actual encode rate

### LINKED ISSUE #43: Disc-capacity budget estimate is disconnected from the encoder's actual fixed bitrate
## Summary

Both the Planner and Overview (after #42) disc-capacity estimates compute an idealized "budgeted" average video bitrate — `usableBytes * 8 / totalDurationSecs`, capped to DVD spec limits — and use that to estimate output size. But the real encode pipeline doesn't consume this computed budget: `plugins/tauri-plugin-spindle-project/src/build/ffmpeg.rs` hardcodes the title video bitrate to a flat `-b:v 6000k` regardless of what the UI's capacity model computed.

(Raised by Codex review on #42: https://github.com/liminal-hq/spindle/pull/42#discussion_r3453683479)

## Impact

When a project's total duration is long enough that the computed budget bitrate drops below 6 Mbps (i.e. `isCapacityConstrained` is true), the UI estimate uses that lower budgeted rate and can report the project as fitting — while the actual build still encodes at the fixed 6 Mbps and may genuinely overflow the disc. Example from the review: a two-hour DVD-5 project would be estimated around its ~4.6 GB usable budget, while the built video alone (at the real fixed 6 Mbps the encoder actually uses) exceeds 5 GB before audio/menus are even added.

In practice, real-world encode output can land either side of the naive flat-rate prediction too — variable bitrate allocation means simple/low-motion content often comes in well under a `-b:v` target — but the UI's capacity model currently has no connection to the rate the pipeline will actually request, in either direction.

## Suggested directions (needs a product/architecture decision, not a quick fix)

1. **Wire the computed budget into the build**: have `generate_build_plan`/the ffmpeg job builder actually use `availableBitsPerSecond` from the capacity estimate as the target `-b:v`/`-maxrate`, so the encode genuinely respects whatever budget the UI promised.
2. **Or, make the estimate honest about the fixed rate**: stop pretending the bitrate adapts to fit, and instead estimate size at the fixed encode rate(s) the pipeline actually uses, surfacing "this project needs N discs" or a clear over-capacity warning instead of a budget that nothing enforces.

This is pre-existing in Planner's original calculation (not introduced by #42) — #42 only made it load-bearing for both Overview and Planner instead of Planner alone, by sharing the same calculation between them.

### PR DESCRIPTION
**Pull request by ScottMorris**
### Build pipeline
- **Make the disc-capacity budget the encoder's actual rate**: the title transcode previously always hardcoded `-b:v 6000k`/`-maxrate 9000k`, ignoring disc capacity entirely. `build_ffmpeg_transcode_command` now takes a computed per-title average bitrate and uses it.
- **Single source of truth for the budget**: ports the capacity calculation into Rust as `build::estimate_disc_capacity`, used by both `generate_build_plan` (to drive the actual encode) and a new `estimate_disc_capacity` Tauri command (for the frontend). The build and the UI estimate can no longer disagree.
- **Per-title bitrate allocation**: the disc-wide budget is distributed across titles honouring `BuildSettings.allocationStrategy` — `duration-weighted` gives every title the same rate (bytes scale with duration), `equal-share` gives every title the same byte budget (short titles get a higher rate). `priority-weighted` falls back to `duration-weighted` until #44 adds per-title weight data.

### Frontend
- `apps/spindle/src/utils/capacity.ts` is now a thin wrapper around the new command (a `useDiscCapacityEstimate` hook + `formatBytes`) instead of a parallel TS implementation of the same math.
- Overview and Planner both consume the hook; Planner's title breakdown now shows the *actual* per-title video bitrate the build will use, not just its duration share.

### Plugin/packaging
- New `estimate_disc_capacity` command added through the plugin's existing guest-js layer (`guest-js/index.ts`), matching the convention adopted in #33 — not a raw `invoke` call from app code.
- Permissions regenerated (`allow-estimate-disc-capacity` added to `default.toml`).

### Follow-up
- Filed #44 for the bigger feature this unblocks but doesn't implement: per-title bitrate weighting, floor/ceiling clamps, an over-floor warning, and manual bitrate pinning.

## Test plan

- [x] `cargo test -p tauri-plugin-spindle-project --lib` — 171 passed, including new `build::capacity` tests (duration-weighted parity, equal-share favouring short titles, no-titles edge case) and new `build::ffmpeg` tests (budgeted rate replaces the old hardcoded 6000k; falls back to 6000k when no budget is available).
- [x] `cargo fmt --check` / `cargo clippy --all-targets -- -D warnings` — clean.
- [x] `npx tsc --noEmit` / `npx vitest run` (71 passed) / `npx prettier --check` — clean.
- [x] Manually traced the default test project's numbers end-to-end (DVD5, one 3600s title, no menus) to confirm the computed budget (~10.2 Mbps raw, capped to 9.8 Mbps DVD spec, then to the encoder's 9 Mbps ceiling) matches what the new ffmpeg test asserts.

Closes #43

### CODE REVIEW

**Comment by ScottMorris** [comment]
@codex review

**Comment by ScottMorris** [comment]
@codex review

File: plugins/tauri-plugin-spindle-project/src/build/capacity.rs [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        .titlesets
+        .iter()
+        .flat_map(|ts| ts.titles.iter())
+        .map(|title| {
+            let duration = title
+                .source_asset_id
+                .as_deref()
+                .and_then(|id| project.assets.iter().find(|a| a.id == id))
+                .and_then(|asset| asset.duration_secs)
+                .unwrap_or(0.0);
```
[the remark this answers is not in the document]
Comment (ScottMorris) [reply plugins/tauri-plugin-spindle-project/src/build/capacity.rs]: Good catch — fixed in 7e674ec. Unknown-duration titles now get `0.0` instead of a capacity-derived rate, which `build_ffmpeg_transcode_command` already treats as "fall back to the safe default" rather than a budgeted rate that would silently ride on the disc-level estimate. Added a regression test (`unknown_duration_titles_do_not_get_a_capacity_derived_bitrate`).

File: plugins/tauri-plugin-spindle-project/src/build/capacity.rs [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        (estimated_output_bytes / capacity_bytes) * 100.0
+    } else {
+        0.0
+    };
+    let is_over_capacity = estimated_output_bytes > usable_bytes;
+
+    let title_bitrates = allocate_title_bitrates(
+        &titles_with_duration,
+        total_duration_secs,
+        available_bits_per_second,
```
[the remark this answers is not in the document]
Comment (ScottMorris) [reply plugins/tauri-plugin-spindle-project/src/build/capacity.rs:130]: Fixed in 7e674ec — made `build_ffmpeg_transcode_command`'s `MAX_VIDEO_RATE_BPS` (9.0 Mbps) `pub(crate)` and clamped the exported per-title rate to it, so Planner/the command can no longer report a rate above what the generated `-b:v` actually ends up being. Added a regression test (`title_bitrates_are_clamped_to_the_encoder_ceiling_not_just_the_dvd_spec_limit`).

**Comment by ScottMorris** [comment]
@codex review

File: plugins/tauri-plugin-spindle-project/src/build/capacity.rs [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+        .sum();
+
+    let safety_margin_bytes = project.build_settings.safety_margin_bytes as f64;
+    let estimated_overhead_bytes = 50_000_000.0 + estimated_menu_bytes;
+    let usable_bytes = capacity_bytes - safety_margin_bytes - estimated_overhead_bytes;
+
+    // NOTE: this budgeted rate is advisory only until it's fed into the build —
+    // see liminal-hq/spindle#43.
+    let raw_bits_per_second = if total_duration_secs > 0.0 {
+        (usable_bytes * 8.0) / total_duration_secs
```
[the remark this answers is not in the document]
Comment (ScottMorris) [reply plugins/tauri-plugin-spindle-project/src/build/capacity.rs]: Good catch — fixed in 6d0a0dc: audio is now estimated per-title (re-encode target, source bitrate for copy-mode, or the silent-fallback default) and subtracted from the usable-byte budget before deriving `available_bits_per_second`. `estimated_output_bytes` now folds the audio bytes in too. Added regression tests for both.

**Comment by ScottMorris** [comment]
@codex review

File: plugins/tauri-plugin-spindle-project/src/build/capacity.rs [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+    title
+        .audio_mappings
+        .iter()
+        .map(|mapping| match mapping.copy_mode {
+            CopyMode::ReEncode => match mapping.output_target {
+                AudioOutputTarget::Ac3 => 448_000.0,
+                AudioOutputTarget::Mp2 => 384_000.0,
+                AudioOutputTarget::Dts => 768_000.0,
+                // Stereo 16-bit/48kHz PCM; LPCM has no fixed rate of its own.
+                AudioOutputTarget::Lpcm => 1_536_000.0,
```
[the remark this answers is not in the document]
Comment (ScottMorris) [reply plugins/tauri-plugin-spindle-project/src/build/capacity.rs]: Good catch — fixed in f819330: LPCM reservation is now derived from the mapped source streams channel count (falling back to stereo only if unknown), instead of assuming stereo. 5.1 LPCM is legal per the DVD spec (up to 8ch/24-bit/96kHz) but burns a lot of mux budget, so the estimate now reflects that honestly rather than under-counting it. Added a regression test comparing stereo vs. 5.1 reservation.

**Comment by ScottMorris** [comment]
@codex review

File: plugins/tauri-plugin-spindle-project/src/build/capacity.rs [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+        (usable_video_bytes * 8.0) / total_duration_secs
+    } else {
+        0.0
+    };
+    let available_bits_per_second = raw_bits_per_second.min(DVD_MAX_VIDEO_RATE_BPS);
+    let is_capacity_constrained = raw_bits_per_second < DVD_MAX_VIDEO_RATE_BPS;
+
+    let estimated_output_bytes = if total_duration_secs > 0.0 {
+        let video_bps = raw_bits_per_second.min(DVD_MAX_MUX_RATE_BPS);
+        ((video_bps * total_duration_secs) / 8.0) + total_audio_bytes
```
[the remark this answers is not in the document]
Comment (ScottMorris) [reply plugins/tauri-plugin-spindle-project/src/build/capacity.rs]: Good catch — fixed in b6af7e8: reordered allocate_title_bitrates ahead of the output-size calc and now sum the already-clamped per-title video bytes plus reserved audio bytes, instead of re-deriving from the raw disc-wide rate capped only to the mux limit. Dropped the now-unused DVD_MAX_MUX_RATE_BPS constant. Added a regression test.

**Comment by ScottMorris** [comment]
@codex review

File: plugins/tauri-plugin-spindle-project/src/build/capacity.rs [hunk 6]
```diff
@@ -0,0 +1,493 @@
...
+            })
+            .collect();
+    }
+
+    // Clamp to the encoder's actual ceiling (not just the looser DVD spec
+    // limit), so the exported rate matches what `build_ffmpeg_transcode_command`
+    // will really request — otherwise Planner/the command can report a rate
+    // (e.g. the DVD spec's 9.8 Mbps) above what the generated `-b:v` ends up
+    // being (clamped to the encoder's 9.0 Mbps ceiling).
+    let capped_rate = available_bits_per_second.min(super::ffmpeg::MAX_VIDEO_RATE_BPS);
```
[the remark this answers is not in the document]
Comment (ScottMorris) [reply plugins/tauri-plugin-spindle-project/src/build/capacity.rs:268]: Good catch — fixed in ecaf198: added a `MUX_RATE_BPS` constant in ffmpeg.rs (replacing the hardcoded `"10080000"` -muxrate literal), and each title's allocated video rate is now also clamped to `MUX_RATE_BPS - that title's own estimated audio bitrate`, so a heavy per-title audio track (multichannel LPCM etc.) cant push the combined stream past the mux ceiling even when the disc-wide video pool alone would not. Added a regression test.

**Comment by ScottMorris** [comment]
@codex review

File: plugins/tauri-plugin-spindle-project/src/build/ffmpeg.rs [hunk 7]
```diff
@@ -7,14 +7,36 @@ use std::path::Path;
...
     title: &Title,
     asset: &Asset,
     disc: &Disc,
     video_info: Option<&VideoStreamInfo>,
+    video_bitrate_bps: f64,
 ) -> Vec<String> {
+    let video_bitrate_bps = if video_bitrate_bps > 0.0 {
+        video_bitrate_bps.min(MAX_VIDEO_RATE_BPS)
+    } else {
+        DEFAULT_VIDEO_BITRATE_BPS
```
[the remark this answers is not in the document]
Comment (ScottMorris) [reply plugins/tauri-plugin-spindle-project/src/build/ffmpeg.rs:37]: Good catch — fixed in 534d8e5: known-duration titles whose computed video budget is exhausted now report ffmpeg::DEFAULT_VIDEO_BITRATE_BPS instead of 0, since thats what build_ffmpeg_transcode_command will actually fall back to. estimated_output_bytes/is_over_capacity now reflect that default rather than understating it. Unknown-duration titles still report 0 deliberately (a different sentinel: no duration data, not just no budget).

File: plugins/tauri-plugin-spindle-project/src/build/capacity.rs [hunk 8]
```diff
@@ -0,0 +1,524 @@
...
+            // back to the heaviest re-encode target (AC3) if unknown.
+            CopyMode::Copy => asset
+                .and_then(|a| {
+                    a.audio_streams
+                        .iter()
+                        .find(|s| s.index == mapping.source_stream_index)
+                })
+                .and_then(|stream| stream.bitrate_bps)
+                .map(|bps| bps as f64)
+                .unwrap_or(448_000.0),
```
[the remark this answers is not in the document]
Comment (ScottMorris) [reply plugins/tauri-plugin-spindle-project/src/build/capacity.rs]: Good catch — fixed in 534d8e5: copy-mode mappings with unknown source bitrate now fall back to the mappings own output_target rate (Ac3/Mp2/Dts/Lpcm) instead of always assuming AC3s 448 kbps, since -c:a copy isnt capped to any particular codecs size.

**Comment by ScottMorris** [comment]
@codex review

File: plugins/tauri-plugin-spindle-project/src/build/capacity.rs [hunk 9]
```diff
@@ -0,0 +1,609 @@
...
+                    // Every title gets the same average rate; bytes naturally
+                    // scale with each title's own duration.
+                    AllocationStrategy::DurationWeighted | AllocationStrategy::PriorityWeighted => {
+                        capped_rate
+                    }
+                    // Every title gets the same total byte budget regardless
+                    // of duration, so short titles get a higher per-second rate.
+                    AllocationStrategy::EqualShare => {
+                        let total_budget_bits = available_bits_per_second * total_duration_secs;
+                        let share_bits = total_budget_bits / titles_with_duration.len() as f64;
```
[the remark this answers is not in the document]
Comment (ScottMorris) [reply plugins/tauri-plugin-spindle-project/src/build/capacity.rs]: Good catch — fixed in 02d6fb6: equal-share now divides by the count of positive-duration titles only, not the full title count, so an unknown-duration title (which gets 0 bps and is excluded from the byte estimate anyway) no longer dilutes every known-duration titles share. Added a regression test.

**Comment by ScottMorris** [comment]
@claude review
